File: src/roar_net_api/api/operations/destruction_neighbourhood.py
# SPDX-FileCopyrightText: © 2025 Authors of the roar-net-api-py project <https://github.com/roar-net/roar-net-api-py/blob/main/AUTHORS>
#
# SPDX-License-Identifier: Apache-2.0

from typing import Protocol, TypeVar

Neighbourhood = TypeVar("Neighbourhood", covariant=True)


class SupportsDestructionNeighbourhood(Protocol[Neighbourhood]):
    def destruction_neighbourhood(self) -> Neighbourhood: ...

File: julia/src/operations/destruction_neighbourhood.jl
# SPDX-FileCopyrightText: © 2025 Authors of the roar-net-api-py project <https://github.com/roar-net/roar-net-api-py/blob/main/AUTHORS>
#
# SPDX-License-Identifier: Apache-2.0
"""
    destruction_neighbourhood(problem::Problem) -> neighbourhood::Neighbourhood

Return the destruction neighbourhood for `problem`, used by destruction-based
algorithms such as iterated local search or large neighbourhood search.
"""
function destruction_neighbourhood(::Problem)::Neighbourhood end
